server/server.go

```diff
@@ -469,9 +473,10 @@ func (s *Server) LoopContext() context.Context {
 
 func (s *Server) startServerLoop(ctx context.Context) {
 	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(ctx)
-	s.serverLoopWg.Add(3)
+	s.serverLoopWg.Add(4)
 	go s.leaderLoop()
 	go s.etcdLeaderLoop()
+	go s.tsoAllocatorLoop()
```

Comment:
I think better not to incldue the `tsoAllocator` in this PR.

Reply:
The reason I include this change is try to show you some ideas about what the later work will be like in order to help review. I'll take away these changes though for this pr.